src/book/03-guides/02-tools/03-webots.mdx

```diff
@@ -96,25 +96,38 @@ sudo apt-get install cmake-curses-gui libprotobuf-dev protobuf-compiler libeigen
    cd NUWebots
    ```
 
-2. Compile the controllers with cmake
+2. Configure the codebase by running
 
-   1. Make a build folder and move into it
+   ```sh
+   ./b configure
+   ```
+
+   <details>
 
-      ```sh
-      mkdir build && cd build
-      ```
+   <summary> Optional Flags </summary>
+   We can configure with several optional flags as follows:
 
-   2. Configure and generate the build files
+   ```sh
+   ./b configure [-i] [-- <flags>]
+   ```
 
-      ```sh
-      cmake ..
-      ```
+   where `-i` runs an interactive configuration with `ccmake`, and `<flags>` can be any combination of the following:
 
-   3. Compile the code
+   | Flag                              | Effect                                                                                                         | Default             |
+   | --------------------------------- | -------------------------------------------------------------------------------------------------------------- | ------------------- |
+   | `-DUSE_ASAN=ON`                   | Compile with [address sanitizer](https://clang.llvm.org/docs/AddressSanitizer.html)                            | `OFF`               |
+   | `-DUSE_UBSAN=ON`                  | Compile with [undefined behaviour sanitizer](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html)      | `OFF`               |
+   | `-DENABLE_CLANG_TIDY=ON`          | Analyse the code being compiled with [clang-tidy](https://clang.llvm.org/extra/clang-tidy/)                    | `OFF`               |
+   | `-DWEBOTS_HOME=path/to/webots`    | Tells the compiler where webots is (usually implicitly set by environment variable)                            | `/usr/local/webots` |
+   | `-DCMAKE_BUILD_TYPE=<build type>` | Tells the compiler what type of build type to use, out of {`Release`, `Debug`, `MinSizeRel`, `RelWithDebInfo`} | `Release`           |
```

Comment:
These are not the **only** flags that can be set. Almost everything that appears in `ccmake` could be specified as a flag on the command line. Maybe reword it to something like this

```
where `-i` runs an interactive configuration with `ccmake`, and `<flags>` can be any valid cmake command line flag. Some common cmake flags are:
```

Reply:
It's a good point. I have said `Some useful cmake flags are` instead though, since these aren't common cmake flags per se, they're just the ones you might commonly use for this repo.